gensim/models/nmf.py

```diff
@@ -422,33 +503,37 @@ def get_document_topics(self, bow, minimum_probability=None,
             kwargs = dict(minimum_probability=minimum_probability)
             return self._apply(corpus, **kwargs)
 
-        v = matutils.corpus2csc([bow], len(self.id2word)).tocsr()
-        h, _ = self._solveproj(v, self._W, v_max=np.inf)
+        v = matutils.corpus2csc([bow], self.num_tokens)
+        h = self._solveproj(v, self._W, v_max=np.inf)
 
         if normalize is None:
             normalize = self.normalize
         if normalize:
-            h.data /= h.sum()
+            h /= h.sum()
 
         return [
-            (idx, proba.toarray()[0, 0])
+            (idx, proba)
             for idx, proba in enumerate(h[:, 0])
-            if not minimum_probability or proba.toarray()[0, 0] > minimum_probability
+            if not minimum_probability or proba > minimum_probability
         ]
 
     def _setup(self, corpus):
         """Infer info from the first document and initialize matrices.
 
         Parameters
         ----------
-        corpus : iterable of list(int, float)
-            Training corpus.
+        corpus : iterable of list of (int, float), optional
+            Training corpus. Contains list of counts of words for every document.
 
         """
-        self._h, self._r = None, None
-        first_doc_it = itertools.tee(corpus, 1)
-        first_doc = next(first_doc_it[0])
-        first_doc = matutils.corpus2csc([first_doc], len(self.id2word))
+        self._h = None
+
+        if isinstance(corpus, scipy.sparse.csc.csc_matrix):
```

Comment:
Is this some special undocumented case? Deserves a comment.

Reply:
Corpus can be csc matrix now, I'll add it to the docstring

Reply:
Fixed.

Reply:
I still see no comment about this code path. **Why** does this method accept two different input types? Is it some optimization?

Reply:
Oops, I added that to the `__init__` docstring, but forgot to mention csc in the `update`. I'll fix it.